bcmp/topology.c

```diff
@@ -95,8 +95,7 @@ static void process_start_topology_event(void) {
   // here we will need to kick off the topo process by looking at our own neighbors and then sending out a request
   CTX.network_topology = new_network_topology();
 
-  static uint8_t num_ports = 0;
-  num_ports = bm_l2_get_num_ports();
+  static const uint8_t num_ports = ADIN2111_PORT_NUM;
```

Comment:
Is this intentional to use an Adin specific macro here? Could this be an artifact(function) of the network device rather than directly using this macro?

Reply:
Great question — I debated this myself and went with the existing macro. Since it is known at compile time I'd rather it be a macro than a function call. I'll go ahead and add a macro in `bm_adin2111.h` right above `ADIN2111_PORT_MASK` — I would name it slightly differently as `ADIN2111_NUM_PORTS`, which I think is a little clearer.

Reply:
Ah, wait a minute, sorry, I see now more clearly something you're trying to point out. L2 and BCMP only know about the NetworkDevice, and it would be better if they didn't have code that assumes the ADIN2111. I'll see how I can clear that up. Maybe a function call would be better. I'll think about it.

Reply:
@matt001k in commit https://github.com/bristlemouth/bm_core/pull/37/commits/979c588c329c4c4d441e69092a2a08bda6183875 I added a `num_ports` function to the network device trait and set context variables in init for L2 and topology. Let me know what you think.

Reply:
That looks awesome! I like it.